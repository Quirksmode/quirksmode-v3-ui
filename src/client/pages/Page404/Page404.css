.Page404 {
  /* Add Styles */
}
