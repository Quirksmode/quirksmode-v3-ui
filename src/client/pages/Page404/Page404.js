import React from 'react';
import { Helmet } from 'react-helmet';
/**
 * 404 Page Component
 *
 * @name Page404
 */
const Page404 = () => (
  <div className="page page404">
    <Helmet>
      <title>Error - 404</title>
      <meta property="og:title" content="Error - 404" />
      <meta name="description" content="404 Page" />
    </Helmet>
    <section className="Page__section Page__section--greyFade Page__hero">
      <div className="Page__sectionInner grid">
        <h1>Error - 404</h1>
      </div>
    </section>
    <div className="Page__sectionInner ContentBlocks__inner grid">
      <div className="Page__content grid__content">
        <p>Oops, looks like this page cannot be found.</p>
      </div>
    </div>
  </div>
);

export default Page404;
