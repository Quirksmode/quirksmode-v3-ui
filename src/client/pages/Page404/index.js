import React from 'react';
import loadable from '@loadable/component';
import Loading from 'components/Loading/Loading';
import ErrorBoundary from 'components/ErrorBoundary/ErrorBoundary';

const Page404 = loadable(() => import(/* webpackPrefetch: true */ './Page404'), {
  fallback: <Loading />
});

const WrappedPage404 = props => (
  <ErrorBoundary>
    <Page404 { ...props } />
  </ErrorBoundary>
);

export default WrappedPage404;
