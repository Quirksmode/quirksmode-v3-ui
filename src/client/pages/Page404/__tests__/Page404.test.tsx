import { findByTestAttr, mountRender } from 'client/tests/testUtils';
import Page404 from '../Page404';

describe('Page404', () => {
  it('matches snapshot and renders correctly', () => {
    const wrapper = mountRender(Page404);
    expect(findByTestAttr(wrapper, 'Page404').length).toEqual(1);
    expect(findByTestAttr(wrapper, 'Page404__title').text()).toEqual(
      'Error - 404'
    );
    expect(wrapper.html).toMatchSnapshot();
  });
});
