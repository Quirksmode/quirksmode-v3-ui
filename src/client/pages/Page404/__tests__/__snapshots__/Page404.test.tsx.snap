// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 matches snapshot and renders correctly 1`] = `[Function]`;
